misk-jooq/build.gradle.kts

```diff
@@ -0,0 +1,98 @@
+dependencies {
+  implementation(Dependencies.guava)
+  implementation(Dependencies.guice)
+  implementation(Dependencies.javaxInject)
+  implementation(Dependencies.jooq)
+  implementation(Dependencies.kotlinxCoroutines)
+  api(Dependencies.kotlinRetry)
+  implementation(project(":misk"))
+  implementation(project(":misk-core"))
+  implementation(project(":misk-inject"))
+  api(project(":misk-jdbc"))
+  api(project(":wisp-logging"))
+
+  testImplementation(Dependencies.assertj)
+  testApi(project(":misk-testing"))
+  testApi(project(":misk-jdbc-testing"))
+
+  // Needed to generate jooq test db classes
+  jooqGenerator(Dependencies.mysql)
+}
+
+afterEvaluate {
+  project.tasks.dokka {
+    outputDirectory = "$rootDir/docs/0.x"
+    outputFormat = "gfm"
+  }
+}
+
+apply(from = "$rootDir/gradle-mvn-publish.gradle")
+
+// Needed to generate jooq test db classes
+buildscript {
+  dependencies {
+    classpath("gradle.plugin.com.boxfuse.client:flyway-release:5.0.2")
+    classpath(Dependencies.mysql)
+  }
+}
+// Needed to generate jooq test db classes
+plugins {
+  id("org.flywaydb.flyway") version "5.2.4"
+  id("nu.studer.jooq") version "5.2"
+}
+
+// Needed to generate jooq test db classes
+flyway {
+  url = "jdbc:mysql://localhost:3500/misk-jooq-test-codegen"
+  user = "root"
+  password = "root"
+  schemas = arrayOf("jooq")
+  locations = arrayOf("filesystem:${project.projectDir}/src/test/resources/db-migrations")
+  sqlMigrationPrefix = "v"
+}
+// Needed to generate jooq test db classes
+jooq {
+  version.set("3.14.8")
+  edition.set(nu.studer.gradle.jooq.JooqEdition.OSS)
+
+  configurations {
+    create("main") {
+      generateSchemaSourceOnCompilation.set(false)
+      jooqConfiguration.apply {
+        jdbc.apply {
+          driver = "com.mysql.cj.jdbc.Driver"
+          url = "jdbc:mysql://localhost:3500/misk-jooq-test-codegen"
+          user = "root"
+          password = "root"
+        }
+        generator.apply {
+          name = "org.jooq.codegen.KotlinGenerator"
+          database.apply {
+            name = "org.jooq.meta.mysql.MySQLDatabase"
+            inputSchema = "jooq"
+            outputSchema = "jooq"
+            includes = ".*"
+            excludes = "(.*?FLYWAY_SCHEMA_HISTORY)|(.*?schema_version)"
+            recordVersionFields = "version"
+          }
+          generate.apply {
+            isJavaTimeTypes = true
+          }
+          target.apply {
+            packageName = "app.cash.backfila.client.misk.jooq.gen"
+            directory   = "${project.projectDir}/src/test/generated/kotlin"
+          }
```

Comment:
What's this for? Is there something in backfila that's used here?

Reply:
It's just the generated code for tests. Will change the package name. Nice spot.